src/weapon.c

```diff
@@ -475,8 +475,8 @@ oselect(struct monst *mtmp, int x)
     for (otmp = mtmp->minvent; otmp; otmp = otmp->nobj) {
         if (otmp->otyp == x
             /* never select non-cockatrice corpses */
-            && !((x == CORPSE || x == EGG)
-                 && !touch_petrifies(&mons[otmp->corpsenm]))
+            && !((x == CORPSE || x == EGG) &&
+		 !(otmp->corpsenm >= LOW_PM && touch_petrifies(&mons[otmp->corpsenm])))
```

Comment:
Also, I think this line also ends up being too long, and should be split up:

```c
            && !((x == CORPSE || x == EGG)
                 && !(otmp->corpsenm >= LOW_PM
                      && touch_petrifies(&mons[otmp->corpsenm])))
```

Actually, I think it might make more sense to organize it like this, since it makes it easier to understand it's going through a list of reasons to exclude objects rather than actual positive requirements for the object being selected, but I understand that may be out of scope for this PR:
```c
    for (otmp = mtmp->minvent; otmp; otmp = otmp->nobj) {
        if (otmp->otyp != x)
            continue;
        /* never select non-cockatrice corpses */
        if ((x == CORPSE || x == EGG)
            && (otmp->corpsenm == NON_PM
                || !touch_petrifies(&mons[otmp->corpsenm])))
            continue;
        if (otmp->oartifact && touch_artifact(otmp, mtmp))
            continue;
        return otmp;
    }
```

Reply:
Thinking about this, should it be?
                || touch_petrifies(&mons[otmp->corpsenm])))
instead of
                || !touch_petrifies(&mons[otmp->corpsenm])))

Reply:
The second one with the '||'?  Based on the comment, the idea is that it's skipping all corpses and eggs except those which can be used to petrify an enemy, so it should skip any '!touch_petrifies' items.  So I think ` if (!touch_petrifies) continue;` would be right.  The original code is difficult to parse because it has the two negations at different depths like `if (!(!touch_petrifies))` but I think it does the same thing.

Reply:
Oh yes, double negative in original. The comment and your explanation makes sense. So the monster wants to grab something of value =). Thanks!